lib/server-jobs.sql

```diff
@@ -66,6 +66,8 @@ WITH updated_jobs AS (
         exit_signal = $exit_signal
     WHERE
         j.id = $job_id
+        -- Ensure we can't finish the same job multiple times.
+        AND status = 'Running'::enum_job_status
```

Comment:
Not that it matters, but the typecast should not be needed here (Postgres can infer types in this situation).

Reply:
Changed in 4cafc0c!